fix bugs

Basarus opened the pull request.

Basarus committed:
fix bugs

Squizzor commented on client/client.js:
```diff
@@ -43,12 +43,22 @@ function removeObjectFromPlayer(player) {
 }
 
-function attachObjectToPlayer(player, boneId, objectName, positionX, positionY, positionZ, rotationX, rotationY, rotationZ) {
+function attachObjectToPlayer(
```
I don't really like this kind of formatting. I would like to keep my style ;)

Squizzor commented on client/client.js:
```diff
@@ -58,38 +68,62 @@ function attachObjectToPlayer(player, boneId, objectName, positionX, positionY,
-        
-        var boneIndex = natives.getPedBoneIndex(player.scriptID, boneId); 
+
+        var boneIndex = natives.getPedBoneIndex(player.scriptID, Number(boneId));
```
Is this cast really necessary? Never had problems without.

Squizzor commented on client/client.js:
```diff
@@ -58,38 +68,62 @@ function attachObjectToPlayer(player, boneId, objectName, positionX, positionY,
-
-            natives.attachEntityToEntity(newObject, player.scriptID, boneIndex, positionX, positionY, positionZ, rotationX, rotationY, rotationZ, 
-                false, false, false, false, 1, true);  
+            console.log(boneId, objectName, positionX, positionY, positionZ, rotationX, rotationY, rotationZ);
```
I would like to keep the code clean from those debug logs. Console outputs will only be made in case of an error.

Squizzor commented on client/client.js:
```diff
@@ -58,38 +68,62 @@ function attachObjectToPlayer(player, boneId, objectName, positionX, positionY,
-            natives.attachEntityToEntity(newObject, player.scriptID, boneIndex, positionX, positionY, positionZ, rotationX, rotationY, rotationZ, 
-                false, false, false, false, 1, true);  
+            console.log(boneId, objectName, positionX, positionY, positionZ, rotationX, rotationY, rotationZ);
+            natives.attachEntityToEntity(
```
Formatting should also not be changed here and I don't like to numerate the parameters.

Squizzor commented on client/client.js:
```diff
@@ -58,38 +68,62 @@ function attachObjectToPlayer(player, boneId, objectName, positionX, positionY,
-        attachObjectToPlayer(player, objectData.boneId, objectData.objectName, objectData.position.x, objectData.position.y, objectData.position.z, 
-            objectData.rotation.x, objectData.rotation.y, objectData.rotation.z);
-    } 
+        attachObjectToPlayer(
```
Formatting shouldn't be changed here as well.

Basarus commented:
my friend, I didn't add anything from myself, I corrected the errors that
came out in the console after the last addition, I didn't touch the rest.
Perhaps the commit was just created this way

Пн, 13 сент. 2021 г. в 00:15, Squizzor ***@***.***>:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In client/client.js
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#discussion_r706913810>
> :
>
> >          outputMessage(e.message);
>      }
>  }
>
>  function attachRegisteredObjectToPlayer(player, objectData) {
>      if (objectData) {
> -        attachObjectToPlayer(player, objectData.boneId, objectData.objectName, objectData.position.x, objectData.position.y, objectData.position.z,
> -            objectData.rotation.x, objectData.rotation.y, objectData.rotation.z);
> -    }
> +        attachObjectToPlayer(
>
> Formatting shouldn't be changed here as well.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#pullrequestreview-752216136>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARZWOSAGZIBJ3MXDDQMBB3TUBUJ5JANCNFSM5DVLYG5Q>
> .
>

Squizzor commented on client/client.js:
```diff
@@ -99,15 +133,15 @@ function attachRegisteredObjectToLocalPlayerSynced(objectName, objectData) {
 }
 
-function detachObjectFromLocalPlayerSynced() {
+function detachObjectFromLocalPlayerSynced(objectName) {
```
The new parameter objectName has no function until this point, cause there is no handling at server side.
At this stage its only possible to hold one object at a time, therefor objectName doesn't need to be passed.
So this should be either removed again or implemented as a whole.

Squizzor commented:
> my friend, I didn't add anything from myself, I corrected the errors that came out in the console after the last addition, I didn't touch the rest. Perhaps the commit was just created this way Пн, 13 сент. 2021 г. в 00:15, Squizzor ***@***.***>:
> […](#)
> ***@***.**** commented on this pull request. ------------------------------ In client/client.js <[#1 (comment)](https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#discussion_r706913810)> : > outputMessage(e.message); } } function attachRegisteredObjectToPlayer(player, objectData) { if (objectData) { - attachObjectToPlayer(player, objectData.boneId, objectData.objectName, objectData.position.x, objectData.position.y, objectData.position.z, - objectData.rotation.x, objectData.rotation.y, objectData.rotation.z); - } + attachObjectToPlayer( Formatting shouldn't be changed here as well. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub <[#1 (review)](https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#pullrequestreview-752216136)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ARZWOSAGZIBJ3MXDDQMBB3TUBUJ5JANCNFSM5DVLYG5Q> .

Could you maybe tell me what were the exact errors + fixes? I really appreciate that you are contributing, but I would like to keep the code clean and fitting to a style that I like ;)

Basarus commented:
After the last update, it is required that the natives are spelled out
completely and strictly typed. In some places, you passed a String instead
of a Number, where some values were missing

пн, 13 сент. 2021 г. в 00:20, Squizzor ***@***.***>:

> my friend, I didn't add anything from myself, I corrected the errors that
> came out in the console after the last addition, I didn't touch the rest.
> Perhaps the commit was just created this way Пн, 13 сент. 2021 г. в 00:15,
> Squizzor *@*.
>
> *>: … <#m_-4090346683823348088_> @.** commented on this pull request.
> ------------------------------ In client/client.js <#1 (comment)
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#discussion_r706913810>>
> : > outputMessage(e.message); } } function
> attachRegisteredObjectToPlayer(player, objectData) { if (objectData) { -
> attachObjectToPlayer(player, objectData.boneId, objectData.objectName,
> objectData.position.x, objectData.position.y, objectData.position.z, -
> objectData.rotation.x, objectData.rotation.y, objectData.rotation.z); - } +
> attachObjectToPlayer( Formatting shouldn't be changed here as well. — You
> are receiving this because you authored the thread. Reply to this email
> directly, view it on GitHub <#1 (review)
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#pullrequestreview-752216136>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/ARZWOSAGZIBJ3MXDDQMBB3TUBUJ5JANCNFSM5DVLYG5Q
> .
>
> Could you maybe tell me what where the exact errors + fixes? I really
> appreciate that you are contributing, but I would to keep the code clean
> and fitting to a style that I like ;)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#issuecomment-917712434>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARZWOSHX5H4IZ3NWGYZM6ZLUBUKR7ANCNFSM5DVLYG5Q>
> .
>

Basarus commented:
import natives from 'natives';
import alt from 'alt';
import registeredObjects from 'client/objects';

const DEBUG_MODE = false;
const OBJECT_RANGE = 30;
const CHECK_INTERVAL = 1000;
const CURSOR_TOGGLE_KEY = 121; // F11

var currentExistingObjects = [];
var cursorActive = false;

function outputMessage(message) {
    console.log('[ObjectAttacher] ' + message);
}

function toggleCursor() {
    try {
        alt.showCursor(!cursorActive);
        alt.toggleGameControls(cursorActive);
        cursorActive = !cursorActive;
    } catch (e) {
        outputMessage(e.message);
    }
}

function getRegisteredObject(objectName) {
    if (registeredObjects[objectName]) {
        return registeredObjects[objectName];
    } else {
        outputMessage('Object is not registered: ' + objectName);
        return null;
    }
}

function removeObjectFromPlayer(player) {
    try {
        var object = currentExistingObjects[player.id];
        if (object && natives.doesEntityExist(object)) {
            natives.detachEntity(object, true, true);
            natives.deleteObject(object);
            currentExistingObjects[player.id] = null;
            // Show weapon again
            natives.setPedCurrentWeaponVisible(player.scriptID, true, true, true, true);
        }
    } catch (e) {
        outputMessage(e.message);
    }
}

function attachObjectToPlayer(
    player,
    boneId,
    objectName,
    positionX,
    positionY,
    positionZ,
    rotationX,
    rotationY,
    rotationZ
) {
    try {
        // Remove existing object (if exists)
        removeObjectFromPlayer(player);

        var hashOfProp = natives.getHashKey(objectName);

        natives.requestModel(hashOfProp);
        const modelLoadInterval = alt.setInterval(() => {
            if (natives.hasModelLoaded(hashOfProp)) {
                alt.clearInterval(modelLoadInterval);
            }
        }, 100);

        var newObject = natives.createObject(hashOfProp, player.pos.x, player.pos.y, player.pos.z, true, true, true);

        // Release memory for model
        natives.setModelAsNoLongerNeeded(hashOfProp);

        var boneIndex = natives.getPedBoneIndex(player.scriptID, Number(boneId));

        if (newObject) {
            // Hide weapon before attaching object
            natives.setPedCurrentWeaponVisible(player.scriptID, false, true, true, true);
            console.log(boneId, objectName, positionX, positionY, positionZ, rotationX, rotationY, rotationZ);
            natives.attachEntityToEntity(
                newObject, // 0
                player.scriptID, // 1
                boneIndex, // 2
                Number(positionX), // 3
                Number(positionY), // 4
                Number(positionZ), // 5
                Number(rotationX), // 6
                Number(rotationY), // 7
                Number(rotationZ), // 8
                false, // 9
                false, // 10
                false, // 11
                false, // 12
                1, // 13
                true // 14
            );

            currentExistingObjects[player.id] = newObject;
        } else {
            outputMessage('Object is null: ' + objectName);
        }
    } catch (e) {
        outputMessage(e.message);
    }
}

function attachRegisteredObjectToPlayer(player, objectData) {
    if (objectData) {
        attachObjectToPlayer(
            player,
            objectData.boneId,
            objectData.objectName,
            objectData.position.x,
            objectData.position.y,
            objectData.position.z,
            objectData.rotation.x,
            objectData.rotation.y,
            objectData.rotation.z
        );
    }
}

function attachRegisteredObjectToLocalPlayerSynced(objectName, objectData) {
    if (objectData) {
        attachRegisteredObjectToPlayer(alt.Player.local, objectData);
        alt.emitServer('objectAttacher:attachedObject', objectName);
    }
}

function detachObjectFromLocalPlayerSynced(objectName) {
    removeObjectFromPlayer(alt.Player.local);
    alt.emitServer('objectAttacher:detachedObject', objectName);
}

function resetAnimationOnLocalPlayer() {
    try {
        natives.clearPedTasks(alt.Player.local.scriptID);
    } catch (e) {
        outputMessage(e.message);
    }
}

function playAnimationOnLocalPlayer(animDictionary, animationName, animationFlag) {
    try {
        if (natives.doesAnimDictExist(animDictionary)) {
            natives.requestAnimDict(animDictionary);

            const animDictLoadInterval = alt.setInterval(() => {
                if (natives.hasAnimDictLoaded(animDictionary)) {
                    alt.clearInterval(animDictLoadInterval);
                }
            }, 100);

            natives.taskPlayAnim(
                alt.Player.local.scriptID,
                animDictionary,
                animationName,
                8.0,
                8.0,
                -1,
                Number(animationFlag),
                1.0,
                false,
                false,
                false
            );
        } else {
            outputMessage('Animation dictionary does not exist');
        }
    } catch (e) {
        outputMessage(e.message);
    }
}

function playAnimationSequenceOnLocalPlayer(enterAnimation, exitAnimation, sequenceFinishedCallback) {
    let enterAnimationIsSet = enterAnimation && enterAnimation.dict && enterAnimation.name;
    let exitAnimationIsSet = exitAnimation && exitAnimation.dict && exitAnimation.name;

    let firstAnimation = null;
    let secondAnimation = null;

    // Only play animations that are completely set
    if (enterAnimationIsSet) {
        firstAnimation = enterAnimation;
        if (exitAnimationIsSet) {
            secondAnimation = exitAnimation;
        }
    } else if (exitAnimationIsSet) {
        firstAnimation = exitAnimation;
    }

    if (firstAnimation) {
        resetAnimationOnLocalPlayer();

        playAnimationOnLocalPlayer(firstAnimation.dict, firstAnimation.name, firstAnimation.flag);

        if (firstAnimation.durationMs && firstAnimation.durationMs > 0) {
            alt.setTimeout(() => {
                if (secondAnimation) {
                    playAnimationOnLocalPlayer(secondAnimation.dict, secondAnimation.name, secondAnimation.flag);

                    if (secondAnimation.durationMs && secondAnimation.durationMs > 0) {
                        alt.setTimeout(() => {
                            resetAnimationOnLocalPlayer();
                            sequenceFinishedCallback();
                        }, secondAnimation.durationMs);
                    }
                } else {
                    resetAnimationOnLocalPlayer();
                    sequenceFinishedCallback();
                }
            }, firstAnimation.durationMs);
        }
    }
}

// Interval for attaching and removing objects from remote players
alt.setInterval(() => {
    try {
        alt.Player.all.forEach(remotePlayer => {
            // Skip local player
            if (remotePlayer.id == alt.Player.local.id) {
                return;
            }

            var objectOfRemotePlayer = remotePlayer.getSyncedMeta('AttachedObject');

            if (objectOfRemotePlayer) {
                var isRemotePlayerInRange =
                    remotePlayer.scriptID && remotePlayer.pos.isInRange(alt.Player.local.pos, OBJECT_RANGE);

                // Object not created yet?
                if (!currentExistingObjects[remotePlayer.id]) {
                    if (isRemotePlayerInRange) {
                        // Attach object to remote player
                        attachRegisteredObjectToPlayer(remotePlayer, getRegisteredObject(objectOfRemotePlayer));
                    }
                } else {
                    // Players is holding object, but is not in range anymore
                    if (!isRemotePlayerInRange) {
                        removeObjectFromPlayer(remotePlayer);
                    }
                }
            } else {
                // Remove object, if player was holding one before
                removeObjectFromPlayer(remotePlayer);
            }
        });
    } catch (e) {
        outputMessage(e.message);
    }
}, CHECK_INTERVAL);

alt.on('objectAttacher:attachObjectAnimated', (objectName, detachObjectAfterAnimation) => {
    var registeredObject = getRegisteredObject(objectName);
    if (registeredObject) {
        attachRegisteredObjectToLocalPlayerSynced(objectName, registeredObject);
        playAnimationSequenceOnLocalPlayer(registeredObject.enterAnimation, registeredObject.exitAnimation, () => {
            if (detachObjectAfterAnimation) {
                detachObjectFromLocalPlayerSynced();
            }
        });
    }
});

alt.onServer('objectAttacher:attachObjectAnimated', (objectName, detachObjectAfterAnimation = true) => {
    var registeredObject = getRegisteredObject(objectName);
    if (registeredObject) {
        attachRegisteredObjectToLocalPlayerSynced(objectName, registeredObject);
        playAnimationSequenceOnLocalPlayer(registeredObject.enterAnimation, registeredObject.exitAnimation, () => {
            if (detachObjectAfterAnimation) {
                detachObjectFromLocalPlayerSynced(objectName);
            }
        });
    }
});

alt.on('objectAttacher:attachObject', objectName => {
    attachRegisteredObjectToLocalPlayerSynced(objectName, getRegisteredObject(objectName));
});

alt.onServer('objectAttacher:attachObject', objectName => {
    attachRegisteredObjectToLocalPlayerSynced(objectName, getRegisteredObject(objectName));
});

alt.on('objectAttacher:detachObject', () => {
    detachObjectFromLocalPlayerSynced();
});

alt.onServer('objectAttacher:detachObject', () => {
    detachObjectFromLocalPlayerSynced();
});

if (DEBUG_MODE) {
    var mainView = new alt.WebView('/resource/client/html/index.html');
    mainView.focus();
    alt.setInterval(() => {
        natives.invalidateIdleCam();
    }, 2000);

    mainView.on('objectAttacher:debug:requestRegisteredObjects', () => {
        mainView.emit('objectAttacher:debug:setRegisteredObjects', registeredObjects);
    });

    mainView.on(
        'objectAttacher:debug:attachObject',
        (objectName, boneId, positionX, positionY, positionZ, rotationX, rotationY, rotationZ) => {
            attachObjectToPlayer(
                alt.Player.local,
                boneId,
                objectName,
                positionX,
                positionY,
                positionZ,
                rotationX,
                rotationY,
                rotationZ
            );
        }
    );

    mainView.on('objectAttacher:debug:detachObject', () => {
        removeObjectFromPlayer(alt.Player.local);
    });

    mainView.on('objectAttacher:debug:playAnimation', animation => {
        playAnimationOnLocalPlayer(animation.dict, animation.name, animation.flag);

        if (animation.durationMs && animation.durationMs > 0) {
            alt.setTimeout(() => {
                resetAnimationOnLocalPlayer();
            }, animation.durationMs);
        }
    });

    mainView.on('objectAttacher:debug:playSequence', (enterAnimation, exitAnimation, detachObjectAfterSequence) => {
        playAnimationSequenceOnLocalPlayer(enterAnimation, exitAnimation, () => {
            if (detachObjectAfterSequence) {
                removeObjectFromPlayer(alt.Player.local);
            }
        });
    });

    mainView.on('objectAttacher:debug:resetAnimation', () => {
        resetAnimationOnLocalPlayer();
    });

    alt.on('consoleCommand', (command, ...args) => {
        if (command === 'objectattacher') {
            mainView.isVisible = !mainView.isVisible;
        }
    });

    alt.on('keyup', function(key) {
        if (key == CURSOR_TOGGLE_KEY) {
            toggleCursor();
        }
    });
}

Basarus commented:
Unfortunately, I can't forward the file to you, since Google does not give
it, just check all your natives and fix it. Otherwise, I did not change the
code. If I'm not mistaken, there was a check with a reference to the bones

пн, 13 сент. 2021 г. в 00:21, Олег Трифонов ***@***.***>:

> After the last update, it is required that the natives are spelled out
> completely and strictly typed. In some places, you passed a String instead
> of a Number, where some values were missing
>
> пн, 13 сент. 2021 г. в 00:20, Squizzor ***@***.***>:
>
>> my friend, I didn't add anything from myself, I corrected the errors that
>> came out in the console after the last addition, I didn't touch the rest.
>> Perhaps the commit was just created this way Пн, 13 сент. 2021 г. в 00:15,
>> Squizzor *@*.
>>
>> *>: … <#m_-777060118972313283_m_-4090346683823348088_> @.** commented on
>> this pull request. ------------------------------ In client/client.js <#1
>> (comment)
>> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#discussion_r706913810>>
>> : > outputMessage(e.message); } } function
>> attachRegisteredObjectToPlayer(player, objectData) { if (objectData) { -
>> attachObjectToPlayer(player, objectData.boneId, objectData.objectName,
>> objectData.position.x, objectData.position.y, objectData.position.z, -
>> objectData.rotation.x, objectData.rotation.y, objectData.rotation.z); - } +
>> attachObjectToPlayer( Formatting shouldn't be changed here as well. — You
>> are receiving this because you authored the thread. Reply to this email
>> directly, view it on GitHub <#1 (review)
>> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#pullrequestreview-752216136>>,
>> or unsubscribe
>> https://github.com/notifications/unsubscribe-auth/ARZWOSAGZIBJ3MXDDQMBB3TUBUJ5JANCNFSM5DVLYG5Q
>> .
>>
>> Could you maybe tell me what where the exact errors + fixes? I really
>> appreciate that you are contributing, but I would to keep the code clean
>> and fitting to a style that I like ;)
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#issuecomment-917712434>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ARZWOSHX5H4IZ3NWGYZM6ZLUBUKR7ANCNFSM5DVLYG5Q>
>> .
>>
>

Squizzor commented:
> Unfortunately, I can't forward the file to you, since Google does not give it, just check all your natives and fix it. Otherwise, I did not change the code. If I'm not mistaken, there was a check with a reference to the bones пн, 13 сент. 2021 г. в 00:21, Олег Трифонов ***@***.***>:
> […](#)
> After the last update, it is required that the natives are spelled out completely and strictly typed. In some places, you passed a String instead of a Number, where some values were missing пн, 13 сент. 2021 г. в 00:20, Squizzor ***@***.***>: > my friend, I didn't add anything from myself, I corrected the errors that > came out in the console after the last addition, I didn't touch the rest. > Perhaps the commit was just created this way Пн, 13 сент. 2021 г. в 00:15, > Squizzor *@*. > > *>: … <#m_-777060118972313283_m_-4090346683823348088_> @.** commented on > this pull request. ------------------------------ In client/client.js <#1 > (comment) > <[#1 (comment)](https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#discussion_r706913810)>> > : > outputMessage(e.message); } } function > attachRegisteredObjectToPlayer(player, objectData) { if (objectData) { - > attachObjectToPlayer(player, objectData.boneId, objectData.objectName, > objectData.position.x, objectData.position.y, objectData.position.z, - > objectData.rotation.x, objectData.rotation.y, objectData.rotation.z); - } + > attachObjectToPlayer( Formatting shouldn't be changed here as well. — You > are receiving this because you authored the thread. Reply to this email > directly, view it on GitHub <#1 (review) > <[#1 (review)](https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#pullrequestreview-752216136)>>, > or unsubscribe > https://github.com/notifications/unsubscribe-auth/ARZWOSAGZIBJ3MXDDQMBB3TUBUJ5JANCNFSM5DVLYG5Q > . > > Could you maybe tell me what where the exact errors + fixes? I really > appreciate that you are contributing, but I would to keep the code clean > and fitting to a style that I like ;) > > — > You are receiving this because you authored the thread. > Reply to this email directly, view it on GitHub > <[#1 (comment)](https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#issuecomment-917712434)>, > or unsubscribe > <https://github.com/notifications/unsubscribe-auth/ARZWOSHX5H4IZ3NWGYZM6ZLUBUKR7ANCNFSM5DVLYG5Q> > . >

Ok, I will take a deeper look at this tomorrow. Thanks for reporting and contributing! ;)

Basarus commented:
Everything is cool, my friend, thank you for your work

Пн, 13 сент. 2021 г. в 00:32, Squizzor ***@***.***>:

> After the last update, it is required that the natives are spelled out
> completely and strictly typed. In some places, you passed a String instead
> of a Number, where some values were missing пн, 13 сент. 2021 г. в 00:20,
> Squizzor *@*.**
>
> *>: … <#m_7749242528116126945_> my friend, I didn't add anything from
> myself, I corrected the errors that came out in the console after the last
> addition, I didn't touch the rest. Perhaps the commit was just created this
> way Пн, 13 сент. 2021 г. в 00:15, Squizzor @. >: …
> <#m_-4090346683823348088_> @.* commented on this pull request.
> ------------------------------ In client/client.js <#1
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1> (comment) <#1
> (comment)
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#discussion_r706913810>>>
> : > outputMessage(e.message); } } function
> attachRegisteredObjectToPlayer(player, objectData) { if (objectData) { -
> attachObjectToPlayer(player, objectData.boneId, objectData.objectName,
> objectData.position.x, objectData.position.y, objectData.position.z, -
> objectData.rotation.x, objectData.rotation.y, objectData.rotation.z); - } +
> attachObjectToPlayer( Formatting shouldn't be changed here as well. — You
> are receiving this because you authored the thread. Reply to this email
> directly, view it on GitHub <#1
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1> (review) <#1
> (review)
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#pullrequestreview-752216136>>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/ARZWOSAGZIBJ3MXDDQMBB3TUBUJ5JANCNFSM5DVLYG5Q
> . Could you maybe tell me what where the exact errors + fixes? I really
> appreciate that you are contributing, but I would to keep the code clean
> and fitting to a style that I like ;) — You are receiving this because you
> authored the thread. Reply to this email directly, view it on GitHub <#1
> (comment)
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#issuecomment-917712434>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/ARZWOSHX5H4IZ3NWGYZM6ZLUBUKR7ANCNFSM5DVLYG5Q
> .
>
> Ah ok, had to test that myself again. Didn't try it with the new update
> yet.
> So there were probably also other pe
>
> Unfortunately, I can't forward the file to you, since Google does not give
> it, just check all your natives and fix it. Otherwise, I did not change the
> code. If I'm not mistaken, there was a check with a reference to the bones
> пн, 13 сент. 2021 г. в 00:21, Олег Трифонов *@*.
>
> *>: … <#m_7749242528116126945_> After the last update, it is required that
> the natives are spelled out completely and strictly typed. In some places,
> you passed a String instead of a Number, where some values were missing пн,
> 13 сент. 2021 г. в 00:20, Squizzor @.*>: > my friend, I didn't add
> anything from myself, I corrected the errors that > came out in the console
> after the last addition, I didn't touch the rest. > Perhaps the commit was
> just created this way Пн, 13 сент. 2021 г. в 00:15, > Squizzor *@*. > > *>:
> … <#m_-777060118972313283_m_-4090346683823348088_> @.** commented on >
> this pull request. ------------------------------ In client/client.js <#1
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1> > (comment) >
> <#1 (comment)
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#discussion_r706913810>>>
> > : > outputMessage(e.message); } } function >
> attachRegisteredObjectToPlayer(player, objectData) { if (objectData) { - >
> attachObjectToPlayer(player, objectData.boneId, objectData.objectName, >
> objectData.position.x, objectData.position.y, objectData.position.z, - >
> objectData.rotation.x, objectData.rotation.y, objectData.rotation.z); - } +
> > attachObjectToPlayer( Formatting shouldn't be changed here as well. — You
> > are receiving this because you authored the thread. Reply to this email >
> directly, view it on GitHub <#1
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1> (review) > <#1
> (review)
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#pullrequestreview-752216136>>>,
> > or unsubscribe >
> https://github.com/notifications/unsubscribe-auth/ARZWOSAGZIBJ3MXDDQMBB3TUBUJ5JANCNFSM5DVLYG5Q
> > . > > Could you maybe tell me what where the exact errors + fixes? I
> really > appreciate that you are contributing, but I would to keep the code
> clean > and fitting to a style that I like ;) > > — > You are receiving
> this because you authored the thread. > Reply to this email directly, view
> it on GitHub > <#1 (comment)
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#issuecomment-917712434>>,
> > or unsubscribe >
> https://github.com/notifications/unsubscribe-auth/ARZWOSHX5H4IZ3NWGYZM6ZLUBUKR7ANCNFSM5DVLYG5Q
> > . >
>
> Ok, I will take a deeper look at this tomorrow. Thanks for reporting and
> contributing! ;)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#issuecomment-917714152>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARZWOSHI7UE5NS2KQOMDGL3UBUL5DANCNFSM5DVLYG5Q>
> .
>

Squizzor commented:
@Basarus 
I applied the crucial changes to my master branch, therefor I will close this pull request.
Thanks for your contribution!

Basarus commented:
You are welcome!

Пн, 13 сент. 2021 г. в 20:38, Squizzor ***@***.***>:

> @Basarus <https://github.com/Basarus>
> I applied the crucial changes to my master branch, therefor I will close
> this pull request.
> Thanks for your contribution!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#issuecomment-918421117>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARZWOSCFZOAASO52YRKKJU3UBYZJZANCNFSM5DVLYG5Q>
> .
>

Basarus commented:
I mean, you're welcome

Пн, 13 сент. 2021 г. в 20:48, Олег Трифонов ***@***.***>:

> You are welcome!
>
> Пн, 13 сент. 2021 г. в 20:38, Squizzor ***@***.***>:
>
>> @Basarus <https://github.com/Basarus>
>> I applied the crucial changes to my master branch, therefor I will close
>> this pull request.
>> Thanks for your contribution!
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Squizzor/SquizzorObjectAttacher/pull/1#issuecomment-918421117>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ARZWOSCFZOAASO52YRKKJU3UBYZJZANCNFSM5DVLYG5Q>
>> .
>>
>
